### PR TITLE
Upgrade Dart and Flutter to latest versions

### PR DESCRIPTION
**Pull request by marandaneto**
- Upgrade dependencies
- fix tooling
- null safety

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -24,10 +24,10 @@ if (flutterVersionName == null) {
 apply plugin: 'com.android.application'
 apply plugin: 'kotlin-android'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
-apply plugin: 'io.sentry.android.gradle'
+// apply plugin: 'io.sentry.android.gradle'
```
Comment (marandaneto) [inline android/app/build.gradle:27]: Deactivated because it requires Java 8, it should be an easy fix.

File: android/app/build.gradle [hunk 2]
```diff
@@ -79,12 +79,12 @@ flutter {
...
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlin_version"
 
 
 }
 
- sentry {
-     includeProguardMapping = true
-     autoUploadProguardMapping = true
- }
+//  sentry {
```
Comment (marandaneto) [inline android/app/build.gradle:87]: Disable `autoInstallation`, Flutter already ships the Android SDK.

File: lib/cart.dart [hunk 3]
```diff
@@ -145,12 +143,8 @@ class _CartViewState extends State<CartView> {
           ]),
         ),
         Padding(padding: EdgeInsets.fromLTRB(10, 0, 0, 0)),
-        FlatButton(
-          height: 40,
-          minWidth: 100,
+        TextButton(
```
Comment (marandaneto) [inline lib/cart.dart:146]: `FlatButton` is deprecated, have to adapt its style.

File: lib/product_details.dart [hunk 4]
```diff
@@ -86,10 +86,11 @@ class _ProductDetailsState extends State<ProductDetails> {
...
                               Sentry.addBreadcrumb(Breadcrumb(
                                   category: "cart.action",
-                                  message: "User added ${args.title} to cart"));
-                              args.callback(args);
+                                  message:
+                                      "User added ${args?.title} to cart"));
+                              args?.callback(args);
 
-                              Scaffold.of(context).showSnackBar(SnackBar(
+                              ScaffoldMessenger.of(context).showSnackBar(SnackBar(
```
Comment (marandaneto) [inline lib/product_details.dart:93]: `Scaffold#showSnackBar` is deprecated

File: lib/product_list.dart [hunk 5]
```diff
@@ -98,7 +97,7 @@ class _ItemListState extends State<ItemsList> {
       builder: (context, cart, child) {
         return Flex(direction: Axis.vertical, children: [
           Expanded(
-              child: FlatButton(
+              child: TextButton(
```
Comment (marandaneto) [inline lib/product_list.dart:100]: `FlatButton` is deprecated

**Review by ndmanvar — APPROVED** [review]

**Comment by ndmanvar** [comment]
thank you @marandaneto 🎉 ❤️

**Pull request merged**